packages/template/__tests__/signature.test.ts

```diff
@@ -153,3 +154,56 @@ interface FullLongSig {
 expectTypeOf<ComponentSignatureArgs<FullLongSig>>().toEqualTypeOf<FullLongSig['Args']>();
 expectTypeOf<ComponentSignatureBlocks<FullLongSig>>().toEqualTypeOf<FullLongSig['Blocks']>();
 expectTypeOf<ComponentSignatureElement<FullLongSig>>().toEqualTypeOf<FullLongSig['Element']>();
+
+// types to simulate the `(element)` helper
+// Issue: https://github.com/typed-ember/glint/issues/610
+type ElementFromTagName<T extends string> = T extends keyof HTMLElementTagNameMap
+  ? HTMLElementTagNameMap[T]
+  : Element;
+type ElementHelperPositional<T extends string> = [name: T];
+type ElementHelperReturn<T extends string> = ComponentLike<{
+  Element: ElementFromTagName<T>;
+  Blocks: { default: [] };
+}>;
+
+interface ElementSignature<T extends string> {
+  Args: {
+    Positional: ElementHelperPositional<T>;
+  };
+  Return: ElementHelperReturn<T> | undefined;
+}
+
+// signature for a component receiving an `(element)`
+interface ElementReceiverSignature<T extends string> {
+  Element: ElementFromTagName<T>;
+  Args: {
+    element: ElementSignature<T>['Return'];
+  };
+  Blocks: {
+    default: [];
+  };
+}
+
+expectTypeOf<ComponentSignatureArgs<ElementReceiverSignature<'div'>>>().toEqualTypeOf<{
+  Named: {
+    element: ElementSignature<'div'>['Return'];
+  };
+  Positional: []
+}>();
+
+expectTypeOf<ComponentSignatureArgs<ElementReceiverSignature<null>>>().toEqualTypeOf<{
```

Comment:
Actually, this line is the problematic part here, I'd say. For what I think needs to be tested, is how the components are invoked, rather than the static definition?

That is, in invocation, this should be `null`. But this is obviously a failure:

> `Type 'null' does not satisfy the constraint 'string'.`

To my knowledge, to properly test this, is to provide _broken_ types. How to do that?